src/Key.php

```diff
@@ -5,18 +5,131 @@
 class Key
 {
     /** @var string */
-    public $id;
+    private $id;
 
     /** @var string */
-    public $secret;
+    private $secret;
+
+    /** @var string */
+    private $certificate;
+
+    /** @var string */
+    private $privateKey;
+
+    /** @var string */
+    private $type;
 
     /**
      * @param string $id
      * @param string $secret
      */
-    public function __construct($id, $secret)
+    public function __construct($id, $item)
     {
         $this->id = $id;
-        $this->secret = $secret;
+        $certificate = $this->getX509Certificate($item);
+        $privateKey = $this->getRSAPrivateKey($item);
+        if (($certificate || $privateKey)) {
+            $this->type = 'rsa';
+            if ($privateKey) {
+                $this->privateKey = $privateKey;
+            };
+            if ($certificate) {
+                $this->certificate = openssl_x509_read($certificate);
+            };
+            if ($certificate && $privateKey) {
+                // openssl_pkey_export($this->privateKey, $privateKey);
+            // openssl_x509_export($this->certificate, $certificate);
+            if (! openssl_x509_check_private_key(
+                $this->certificate, $this->privateKey)
+                ) {
+                throw new KeyException("Supplied Certificate and Key are not related");
+            }
+            };
+        } else {
+            $this->type = 'secret';
+            $this->secret = $item;
+            $publicKey = null;
+            $privateKey = null;
+        };
+    }
+
+    private function getRSAPrivateKey($object)
+    {
+        $key = null;
+        if (is_array($object)) {
+            foreach ($object as $item) {
+                $privateKey = Key::getRSAPrivateKey($item);
+                if ($privateKey) {
+                    return $privateKey;
+                }
+            };
+        } else {
+            try {
+                $privateKey = openssl_get_privatekey($object);
+            } catch (\Exception $e) {
+                $privateKey = null;
+            };
```

Comment:
Any reason we want to catch exceptions here? And if so can we be more explicit about what exception?